### PR TITLE
Keywords docs

### PR DESCRIPTION
**Pull request by abadger**
##### SUMMARY

The main fix here is  enhancing playbook object dumping
    
    * After reading the caveats in the doc, it became apparent that this was
      information about the ansible.playbook code which sometimes maps to
      the directives allowed in a playbook but sometimes not.  Moved it to
      dev_guide for developing the core and rename filename and title to
      reflect this.  Perhaps later we can use this same data to describe the
      playbook keywords that the users have access to.
    
    * Turn links in the keyword description into seealso entries in the rst.

Second fix is cleaning up trailing whitespace in plugin_formatter.

##### ISSUE TYPE
 - Docs Pull Request

##### ANSIBLE VERSION

```
devel 2.4
```

\cc @dharmabumstead  @bcoca

### CODE REVIEW

**Comment by abadger** [comment]
bcoca approved.  Waiting on dharmabumstead for final approval and cherry-pick to 2.4

File: docs/docsite/keyword_desc.yml
Comment (dharmabumstead) [inline docs/docsite/keyword_desc.yml]: “Its”, not “it’s”.

**Review by dharmabumstead — APPROVED** [review]
OK with some edits. Thanks @abadger

**Pull request merged**
